app/Http/Requests/UpdateUserRequest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Validation\Rule;
+use Illuminate\Foundation\Http\FormRequest;
+
+class UpdateUserRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules()
+    {
+        return [    'name' => ['required','min:3'],
+        'password'=> 'required | confirmed | min:6',
+     'email' => ['required', 'email', Rule::unique('users','email')],
+     'role_id' => 'required'
+    ];
+    }
+
+    public function FiltredAttributes() {
+
+        $validatedData = $this->validated();
+        
+         // Hash password
+         $validatedData['password'] = bcrypt($this['password']);
```

Comment:
what you did here is correct but you can also use the laravel mututor, for example you will never ever save a password explicitly so instead of hashing each time inside your classes you can do it one time in your user model , check that link [https://laravel.com/docs/9.x/eloquent-mutators](url)